structure_file_input_snippet crashing with g++-8

g++-8.3.0, gcc-8.3.0

The full error log is:
```sh
In file included from /home/joshjayk/development/seqan3/include/seqan3/io/structure_file/output.hpp:41,
                 from /home/joshjayk/development/seqan3/test/snippet/io/structure_file/structure_file_input.cpp:6:
/home/joshjayk/development/seqan3/include/seqan3/range/view/get.hpp: In instantiation of ‘<lambda(auto:14&&)> [with auto:14 = seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >&]’:
/home/joshjayk/development/seqan3/submodules/range-v3/include/range/v3/functional/invoke.hpp:116:14:   required by substitution of ‘template<class F, class ... Args> constexpr decltype (static_cast<F&&>(f)(static_cast<Args&>(ranges::invoke_fn::operator()::args) ...)) ranges::invoke_fn::operator()(F&&, Args&& ...) const [with F = <lambda(auto:14&&)>&; Args = {seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >&}]’
/home/joshjayk/development/seqan3/submodules/range-v3/include/range/v3/functional/invoke.hpp:151:24:   required by substitution of ‘template<class _Fn, class ... _Args> static std::__result_of_success<decltype (declval<_Fn>()((declval<_Args>)()...)), std::__invoke_other> std::__result_of_other_impl::_S_test(int) [with _Fn = const volatile ranges::view::transform_fn&; _Args = {std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&, const volatile<lambda(auto:14&&)>&}]’
/usr/include/c++/8/type_traits:2294:55:   required from ‘struct std::__result_of_impl<false, false, const volatile ranges::view::transform_fn&, std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&, const volatile<lambda(auto:14&&)>&>’
/usr/include/c++/8/type_traits:2299:12:   required from ‘struct std::__invoke_result<const volatile ranges::view::transform_fn&, std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&, const volatile<lambda(auto:14&&)>&>’
/usr/include/c++/8/type_traits:2312:12:   required from ‘class std::result_of<const volatile ranges::view::transform_fn&(std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > >&, const volatile<lambda(auto:14&&)>&)>’
/usr/include/c++/8/functional:444:72:   required by substitution of ‘template<class _Functor, class ... _Bound_args> template<class _Fn, class _CallArgs, class ... _BArgs> using _Res_type_impl = typename std::result_of<_Fn&(std::_Bind<_Functor(_Bound_args ...)>::_Mu_type<_BArgs, _CallArgs>&& ...)>::type [with _Fn = std::add_cv<ranges::view::transform_fn>::type; _CallArgs = std::tuple<std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&>; _BArgs = {std::add_cv<std::_Placeholder<1> >::type, std::add_cv<<lambda(auto:14&&)> >::type}; _Functor = ranges::view::transform_fn; _Bound_args = {std::_Placeholder<1>, <lambda(auto:14&&)>}]’
/usr/include/c++/8/functional:457:46:   required by substitution of ‘template<class _Functor, class ... _Bound_args> template<class _CallArgs, template<class _CallArgs, template<class> class __cv_quals> template<class _Functor, class ... _Bound_args> template<class> class __cv_quals> using _Res_type_cv = std::_Bind<_Functor(_Bound_args ...)>::_Res_type_impl<typename __cv_quals<typename std::enable_if<(bool)((std::tuple_size<_Tuple>::value + 1)), _Functor>::type>::type, _CallArgs, typename __cv_quals<_Bound_args>::type ...> [with _CallArgs = std::tuple<std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&>; __cv_quals = std::add_cv; _Functor = ranges::view::transform_fn; _Bound_args = {std::_Placeholder<1>, <lambda(auto:14&&)>}]’
/usr/include/c++/8/functional:518:9:   required by substitution of ‘template<class ... _Args, class _Result> _Result std::_Bind<ranges::view::transform_fn(std::_Placeholder<1>, <lambda(auto:14&&)>)>::operator()<_Args ..., _Result>(_Args&& ...) const volatile [with _Args = {std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&}; _Result = <missing>]’
/home/joshjayk/development/seqan3/submodules/range-v3/include/range/v3/functional/invoke.hpp:116:14:   required by substitution of ‘template<class F, class ... Args> constexpr decltype (static_cast<F&&>(f)(static_cast<Args&>(ranges::invoke_fn::operator()::args) ...)) ranges::invoke_fn::operator()(F&&, Args&& ...) const [with F = ranges::detail::pipeable_binder<std::_Bind<ranges::view::transform_fn(std::_Placeholder<1>, <lambda(auto:14&&)>)> >&; Args = {std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&}]’
/home/joshjayk/development/seqan3/submodules/range-v3/include/range/v3/functional/concepts.hpp:25:5:   required by substitution of ‘template<class Arg, class Pipe> decltype (ranges::pipeable_access::impl<Pipe>::pipe(declval<Arg>(), declval<Pipe&>())) ranges::operator|(Arg&&, Pipe) requires !(is_pipeable_v<Arg>) and is_pipeable_v<Pipe> [with Arg = std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > >&; Pipe = ranges::view::view<ranges::detail::pipeable_binder<std::_Bind<ranges::view::transform_fn(std::_Placeholder<1>, <lambda(auto:14&&)>)> > >]’
/home/joshjayk/development/seqan3/test/snippet/io/structure_file/structure_file_input.cpp:150:57:   required from here
/home/joshjayk/development/seqan3/include/seqan3/range/view/get.hpp:70:19: error: static assertion failed: You may only pass ranges to view::get whose reference_t models the TupleLike.
     static_assert(TupleLike<decltype(in)>,
                   ^~~~~~~~~~~~~~~~~~~~~~~
/home/joshjayk/development/seqan3/include/seqan3/range/view/get.hpp: In instantiation of ‘<lambda(auto:14&&)> [with auto:14 = seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >&]’:
/home/joshjayk/development/seqan3/submodules/range-v3/include/range/v3/functional/invoke.hpp:116:14:   required by substitution of ‘template<class F, class ... Args> constexpr decltype (static_cast<F&&>(f)(static_cast<Args&>(ranges::invoke_fn::operator()::args) ...)) ranges::invoke_fn::operator()(F&&, Args&& ...) const [with F = <lambda(auto:14&&)>&; Args = {seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >&}]’
/home/joshjayk/development/seqan3/submodules/range-v3/include/range/v3/functional/invoke.hpp:151:24:   required by substitution of ‘template<class _Fn, class ... _Args> static std::__result_of_success<decltype (declval<_Fn>()((declval<_Args>)()...)), std::__invoke_other> std::__result_of_other_impl::_S_test(int) [with _Fn = const volatile ranges::view::transform_fn&; _Args = {std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&, const volatile<lambda(auto:14&&)>&}]’
/usr/include/c++/8/type_traits:2294:55:   required from ‘struct std::__result_of_impl<false, false, const volatile ranges::view::transform_fn&, std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&, const volatile<lambda(auto:14&&)>&>’
/usr/include/c++/8/type_traits:2299:12:   required from ‘struct std::__invoke_result<const volatile ranges::view::transform_fn&, std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&, const volatile<lambda(auto:14&&)>&>’
/usr/include/c++/8/type_traits:2312:12:   required from ‘class std::result_of<const volatile ranges::view::transform_fn&(std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > >&, const volatile<lambda(auto:14&&)>&)>’
/usr/include/c++/8/functional:444:72:   required by substitution of ‘template<class _Functor, class ... _Bound_args> template<class _Fn, class _CallArgs, class ... _BArgs> using _Res_type_impl = typename std::result_of<_Fn&(std::_Bind<_Functor(_Bound_args ...)>::_Mu_type<_BArgs, _CallArgs>&& ...)>::type [with _Fn = std::add_cv<ranges::view::transform_fn>::type; _CallArgs = std::tuple<std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&>; _BArgs = {std::add_cv<std::_Placeholder<1> >::type, std::add_cv<<lambda(auto:14&&)> >::type}; _Functor = ranges::view::transform_fn; _Bound_args = {std::_Placeholder<1>, <lambda(auto:14&&)>}]’
/usr/include/c++/8/functional:457:46:   required by substitution of ‘template<class _Functor, class ... _Bound_args> template<class _CallArgs, template<class _CallArgs, template<class> class __cv_quals> template<class _Functor, class ... _Bound_args> template<class> class __cv_quals> using _Res_type_cv = std::_Bind<_Functor(_Bound_args ...)>::_Res_type_impl<typename __cv_quals<typename std::enable_if<(bool)((std::tuple_size<_Tuple>::value + 1)), _Functor>::type>::type, _CallArgs, typename __cv_quals<_Bound_args>::type ...> [with _CallArgs = std::tuple<std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&>; __cv_quals = std::add_cv; _Functor = ranges::view::transform_fn; _Bound_args = {std::_Placeholder<1>, <lambda(auto:14&&)>}]’
/usr/include/c++/8/functional:518:9:   required by substitution of ‘template<class ... _Args, class _Result> _Result std::_Bind<ranges::view::transform_fn(std::_Placeholder<1>, <lambda(auto:14&&)>)>::operator()<_Args ..., _Result>(_Args&& ...) const volatile [with _Args = {std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&}; _Result = <missing>]’
/home/joshjayk/development/seqan3/submodules/range-v3/include/range/v3/functional/invoke.hpp:116:14:   required by substitution of ‘template<class F, class ... Args> constexpr decltype (static_cast<F&&>(f)(static_cast<Args&>(ranges::invoke_fn::operator()::args) ...)) ranges::invoke_fn::operator()(F&&, Args&& ...) const [with F = ranges::detail::pipeable_binder<std::_Bind<ranges::view::transform_fn(std::_Placeholder<1>, <lambda(auto:14&&)>)> >&; Args = {std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> >, std::allocator<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > > >&}]’
/home/joshjayk/development/seqan3/submodules/range-v3/include/range/v3/functional/concepts.hpp:25:5:   required by substitution of ‘template<class Arg, class Pipe> decltype (ranges::pipeable_access::impl<Pipe>::pipe(declval<Arg>(), declval<Pipe&>())) ranges::operator|(Arg&&, Pipe) requires !(is_pipeable_v<Arg>) and is_pipeable_v<Pipe> [with Arg = std::vector<seqan3::structured_rna<seqan3::rna5, seqan3::wuss<51> > >&; Pipe = ranges::view::view<ranges::detail::pipeable_binder<std::_Bind<ranges::view::transform_fn(std::_Placeholder<1>, <lambda(auto:14&&)>)> > >]’
/home/joshjayk/development/seqan3/test/snippet/io/structure_file/structure_file_input.cpp:151:62:   required from here
/home/joshjayk/development/seqan3/include/seqan3/range/view/get.hpp:70:19: error: static assertion failed: You may only pass ranges to view::get whose reference_t models the TupleLike.
```

--- a/include/seqan3/alphabet/nucleotide/rna15.hpp
+++ b/include/seqan3/alphabet/nucleotide/rna15.hpp
@@ -72,6 +72,9 @@ public:
 
     //!\brief Allow implicit construction from dna/rna of the same size.
     constexpr rna15(dna15 const & r) noexcept
+#if SEQAN3_WORKAROUND_GCC_90897
+        requires true
+#endif
     {
         assign_rank(r.to_rank());
     }

--- a/include/seqan3/alphabet/nucleotide/rna5.hpp
+++ b/include/seqan3/alphabet/nucleotide/rna5.hpp
@@ -70,6 +70,9 @@ public:
 
     //!\brief Allow implicit construction from dna/rna of the same size.
     constexpr rna5(dna5 const & r) noexcept
+#if SEQAN3_WORKAROUND_GCC_90897
+        requires true
+#endif
     {
         assign_rank(r.to_rank());
     }

